Validation/EventGenerator/plugins/LheWeightValidation.cc

```diff
@@ -0,0 +1,336 @@
+#include "Validation/EventGenerator/interface/LheWeightValidation.h"
+
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "Validation/EventGenerator/interface/DQMHelper.h"
+#include "Validation/EventGenerator/interface/HepMCValidationHelper.h"
+
+using namespace edm;
+
+LheWeightValidation::LheWeightValidation(const edm::ParameterSet& iPSet) {
+  lheLabel_ = iPSet.getParameter<edm::InputTag>("lheProduct"),
+  genParticleToken_ = consumes<reco::GenParticleCollection>(iPSet.getParameter<edm::InputTag>("genParticles"));
+  lheEvtToken_ = consumes<LHEEventProduct>(lheLabel_);
+  lheRunToken_ = consumes<LHERunInfoProduct, edm::InRun>(lheLabel_);
+  genJetToken_ = consumes<reco::GenJetCollection>(iPSet.getParameter<edm::InputTag>("genJets"));
+  dumpLHEheader_ = iPSet.getParameter<bool>("dumpLHEheader");
+  nScaleVar_ = iPSet.getParameter<int>("nScaleVar");
+  idxPdfStart_ = iPSet.getParameter<int>("idxPdfStart");
+  idxPdfEnd_ = iPSet.getParameter<int>("idxPdfEnd");
+  leadLepPtRange_ = iPSet.getParameter<double>("leadLepPtRange");
+  leadLepPtNbin_ = iPSet.getParameter<int>("leadLepPtNbin");
+  leadLepPtCut_ = iPSet.getParameter<double>("leadLepPtCut");
+  lepEtaCut_ = iPSet.getParameter<double>("lepEtaCut");
+  rapidityRange_ = iPSet.getParameter<double>("rapidityRange");
+  rapidityNbin_ = iPSet.getParameter<int>("rapidityNbin");
+  jetPtCut_ = iPSet.getParameter<double>("jetPtCut");
+  jetEtaCut_ = iPSet.getParameter<double>("jetEtaCut");
+  nJetsNbin_ = iPSet.getParameter<int>("nJetsNbin");
+  jetPtRange_ = iPSet.getParameter<double>("jetPtRange");
+  jetPtNbin_ = iPSet.getParameter<int>("jetPtNbin");
+
+  nPdfVar_ = idxPdfEnd_ - idxPdfStart_ + 1;
+}
+
+LheWeightValidation::~LheWeightValidation() {}
+
+void LheWeightValidation::bookHistograms(DQMStore::IBooker& iBook, edm::Run const& iRun, edm::EventSetup const&) {
+  // check LHE product exists
+  edm::Handle<LHERunInfoProduct> lheInfo;
+  iRun.getByLabel(lheLabel_, lheInfo);
```

Comment:
`getByLabel` is deprecated, see https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideEDMGetDataFromEvent#get_ByLabel_deprecated

Reply:
Hi @perrotta, thanks for the prompt and comprehensive review. I've addressed all of your comments except this one since I couldn't find a workaround for it. I used `getByToken` elsewhere except this part since I need to retrieve `LHERunInfoProduct` at the `beginRun` stage here (`getByToken` throws an exception unless you do this in the `endRun` stage) - which similar approach is also being used in the NanoAOD package:

https://github.com/cms-sw/cmssw/blob/f84137335f59a16d4cf9979547f0d77180f4e7e1/PhysicsTools/NanoAOD/plugins/GenWeightsTableProducer.cc#L551-L558

Do you know any workaround for this issue while avoiding using `getByLabel`?

Reply:
Thank you @SanghyunKo , I did not know about that issue.
I have no suggestions: unless @Dr15Jones have some, could you please at least add the same comment as in the snippet above to justify the usage of `getByLabel`?

Reply:
@perrotta no problem! I've added the comment.